transformer_lens/HookedTransformer.py

```diff
@@ -133,8 +140,24 @@ def __init__(
             self.init_weights()
 
         if move_to_device:
-            # Move the model to the relevant device, suppress the logging message
-            self.to(self.cfg.device, print_details=False)
+            if self.cfg.n_devices == 1:
+                # Move the model to the relevant device, suppress the logging message
+                self.to(self.cfg.device, print_details=False)
+            else:
+                # We load the devices in a pipeline manner - the first device gets the embed and pos_embed layers and the first n_layers // n_devices blocks,
```

Comment:
Huh, I *was* going to object that this would overload the first and last device, for for GPT-J and up, d_model is >=4K, so a block is 12 * d_model >= 50K width, vs 50K width for the embedding, so it's kinda chill

Reply:
I was also considering keeping the token and position embeddings, final layer norm, and unembed all on the same gpu (probably the rank 0 one) and then splitting the layers across the remaining GPUs. This would be cleaner in a few ways since inputs and outputs would all be on the same device. Lmk what you think of that.